src/libraries/System.Net.NameResolution/src/System/Net/Dns.cs

```diff
@@ -466,7 +524,46 @@ private static Task GetHostEntryOrAddressesCoreAsync(string hostName, bool justR
             if (NameResolutionPal.SupportsGetAddrInfoAsync && ipAddress is null)
             {
                 ValidateHostName(hostName);
-                return NameResolutionPal.GetAddrInfoAsync(hostName, justAddresses);
+
+                if (NameResolutionTelemetry.IsEnabled)
+                {
+                    ValueStopwatch stopwatch = NameResolutionTelemetry.Log.BeforeResolution(hostName);
+
+                    Task coreTask;
+                    try
+                    {
+                        coreTask = NameResolutionPal.GetAddrInfoAsync(hostName, justAddresses);
+                    }
+                    catch when (LogFailure(stopwatch))
+                    {
+                        Debug.Fail("LogFailure should return false");
+                        throw;
+                    }
+
+                    // Only pay for the overhead if events will be fired
+                    if (stopwatch.IsActive)
```

Comment:
I guess we check this just because someone could have toggle off the telemetry in between https://github.com/dotnet/runtime/pull/38409/files#diff-434fc33b2da88cc8b47b46b328d07222R528 and here?

Reply:
This was a leftover check from an invalid optimization. It is redundant, removed.